### PR TITLE
First draft of your first rule, add rule validator

### PR DESCRIPTION
**Pull request by tyler-mairose-sp**

### CODE REVIEW

File: products/idn/docs/identity-now/rules/guides/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+id: guides
+title: Rule Guides
+pagination_label: Guides
+sidebar_label: Guides
+sidebar_position: 1
+sidebar_class_name: Rules
+keywords: ['rules', 'guides']
+description: Transform Guides
```
Comment (colin-mckibben-sp) [inline products/idn/docs/identity-now/rules/guides/index.md:9]: Transform or rule guides?

File: products/idn/docs/identity-now/rules/rule_validator.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+This utility is a Java based application and requires **Java SE Development Kit (JDK) 11** or higher to run.
+
+:::info
+Please note that the previous version used JDK 8 (1.8), if you do not have the JDK 11 or above then this must be downloaded and installed first
+:::
+
+## Support / Features
+
+This utility is developed by SailPoint Technology Services, and its usage is covered via SailPoint Expert Services. If you have any issues, bugs, or feature requests, please submit them to SailPoint Expert Services.
```
Comment (colin-mckibben-sp) [inline products/idn/docs/identity-now/rules/rule_validator.md:47]: Can you link to expert services?

File: products/idn/docs/identity-now/rules/guides/your_first_rule.md [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+In this guide you'll learn the end to end process of writing a cloud rule to generate a username to be used during account creation.
+
+- [Attribute generator rule overview](#attribute-generator-rule)
+- [Username requirements](#username-requirements)
+- [Writing the rule](#writing-the-rule)
+- [Validating the rule](#validate-the-rule)
+- [Submitting for rule review](#submitting-for-rule-review)
+
+## Attribute Generator rule
```
Comment (colin-mckibben-sp) [inline products/idn/docs/identity-now/rules/guides/your_first_rule.md:24]: Please use title case for all section headers.  In this instance, "rule" needs to be capitalized.  Please review all other section headers for title case.

**Review by colin-mckibben-sp — CHANGES_REQUESTED** [review]
Some minor things to change.  I also noticed many grammatical errors.  Please have @james-haytko-sp review this as well for grammar, spelling, and style.

**Review by james-haytko-sp — APPROVED** [review]
I made some edits. Looks good!

**Pull request merged**